proof/refine/ARM_HYP/CSpace1_R.thy

```diff
@@ -4723,12 +4725,14 @@ lemma mdb_inv_preserve_updateCap:
   "\<lbrakk>m slot = Some cte;isUntypedCap (cteCap cte)\<rbrakk> \<Longrightarrow>
    mdb_inv_preserve m (modify_map m slot
      (cteCap_update (\<lambda>_. capFreeIndex_update (\<lambda>_. index) (cteCap cte))))"
-  apply (clarsimp simp:mdb_inv_preserve_def modify_map_dom isCap_simps modify_map_def split:if_splits)
+  apply (clarsimp simp: mdb_inv_preserve_def modify_map_dom gen_isCap_simps modify_map_def
+                  split: if_splits)
   apply (intro conjI impI allI)
-    apply fastforce
-    apply (simp add:sameRegionAs_update_untyped)
-    apply (rule ext,simp add:sameRegionAs_update_untyped')
-    apply (simp add:mdb_next_def split:if_splits)
+      apply fastforce
+     apply (simp add: ARM_HYP.isCap_defs)
```

Comment:
Out of curiosity, any reason why this is `isCap_defs` instead of `isCap_simps` like all the other cases? And also, should it be marked `FIXME arch-split` like those other cases?

Reply:
I have not a clue. To give context, defs look like this:
```
isZombie ?v ≡ case ?v of capability.Zombie v0 v1 v2 ⇒ True | _ ⇒ False
```
while simps look like this:
```
isZombie ?v = (∃v0 v1 v2. ?v = capability.Zombie v0 v1 v2)
```

And the FIXME... yeah, for a moment I thought this file was in CRefine. Since it's in Refine, it should get one. Will do.

Reply:
If it gets the proof done, either is fine. Agreed with the FIXME.